pg_shard.c

```diff
@@ -741,14 +894,170 @@ RowAndColumnFilterQuery(Query *query, List *remoteRestrictList, List *localRestr
 	fromExpr->quals = (Node *) make_ands_explicit((List *) remoteRestrictList);
 	fromExpr->fromlist = QueryFromList(rangeTableList);
 
+	if (remoteQueryHasAggregates)
+	{
+		distributedQuery->hasAggs = true;
+		distributedQuery->groupClause = list_copy(query->groupClause);
+		distributedQuery->targetList =
+			BuildAggregatedDistributedTargetList(query, localRestrictList);
```

Comment:
These dual line assignments are an eyesore. Any way to assign to a local variable and then assign to the field later?